lib/resources/auth/token_repository.dart

```diff
@@ -0,0 +1,125 @@
+import 'package:flutter_secure_storage/flutter_secure_storage.dart';
+import 'package:synchronized/synchronized.dart';
+
+import 'package:wastexchange_mobile/util/jwt_utils.dart';
+
+/// Abstract implementation of TokenRepository.
+///
+/// TokenRepository can be used as base class for implementing the Authentication Token Functionality.
+abstract class TokenRepository {
+  Future<void> setToken(String token);
+
+  Future<String> getToken();
+
+  Future<void> setRefreshToken(String token);
+
+  Future<String> _getRefreshToken();
+
+  Future<void> deleteToken();
+
+  Future<bool> isTokenExpired();
+
+  Future<bool> refreshToken();
+
+  Future<bool> isAuthorized();
+}
+
+/// TokenRepository class specifically designed to handle JWT Token by validating JWT Token, Checking token expired, etc...
+/// This uses a singleton pattern to ensure only one instance is available.
+class JWTTokenRepository implements TokenRepository {
```

Comment:
This class could be ```TokenRepository```

Reply:
Changed to TokenRepository and removed unwanted implementations